presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -659,6 +651,14 @@ void PrestoServer::registerCustomOperators() {
       std::make_unique<operators::ShuffleReadTranslator>());
 }
 
+void PrestoServer::registerAllFunctions() {
```

Comment:
Let's create a base class PrestoServerBase that doesn't have any presto or spark-specific logic and then move engine-specific logic to derived classes.

Reply:
Similar to the `etc` case, isn't it expected for Presto to be perceived as the default case (hence `PrestoServer`) & anything else (Spark) as a special case? :)

Reply:
I think it makes sense to Presto and Presto-on-Spark to be peers. One is for interactive queries, the other for batch.